packages/experiments/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import {
+  IStyle
+} from '../../Styling';
+import { IStyleFunction } from '../../Utilities';
+
+export interface IShimmer {
+
+}
+
+/**
+ * Shimmer component props.
+ */
+export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
+  /**
+   * Optional callback to access the IShimmer interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IShimmer) => void;
+
+  /**
+   * Sets the width of the shimmer wave container in percentages.
+   * @default 100%
+   */
+  width?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
+
+  /**
+   * Elements to render in one line of the Shimmer.
+   */
+  lineElements?: Array<ICircle | IGap | ILine>;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules.
+   */
+  getStyles?: IStyleFunction<IShimmerStyleProps, IShimmerStyles>;
+}
+
+export interface IShimmerElement {
+  /**
+   * Required for every element you intend to use.
+   */
+  type: ShimmerElementType;
+
+  /**
+   * The hight of the element (ICircle, ILine, IGap).
+   * Read more details for each specific element.
+   */
+  height?: number;
+
+  /**
+   * The width of the element (ICircle, ILine, IGap).
+   * Read more details for each specific element.
+   */
+  width?: number;
+
+  /**
+   * The vertical alignemt of the element (ICircle, ILine, IGap).
+   * @default center
+   */
+  verticalAlign?: ShimmerElementVerticalAlign;
+}
+
+export interface ILine extends IShimmerElement {
+  /**
+   * Sets the height of the rectangle.
+   * The value will represent the height in pixels.
+   * @default 16
+   */
+  height?: number;
+
+  /**
+   * If not provided the line will the take the remaining space to fill the shimmer container.
+   * The value provided will represent the width as '%' relative to the shimmer container.
+   */
+  width?: number;
+}
+
+export interface ICircle extends IShimmerElement {
+  /**
+   * Sets the height of the circle.
+   * The value will represent the height in pixels.
+   * @default 24
+   */
+  height?: number;
+
+  /**
+   * Width of the circle element if provided will be ignored.
```

Comment:
Why even have this?

Reply:
Documentation reasons. But I guess if we remove it completely users would not even know they can use it.))

Reply:
It was present more for documentation purposes but I guess if removed the user will not no that it can be used and if he somehow uses it he will notice it does not have any effect.